src/KubernetesClient/GenericClient.cs

```diff
@@ -0,0 +1,78 @@
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    public class GenericClient
+    {
+        internal class TweakApiHandler : DelegatingHandler
+        {
+            protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage msg, CancellationToken cancel)
+            {
+                msg.RequestUri = new Uri(msg.RequestUri, msg.RequestUri.AbsolutePath.Replace("/apis//", "/api/"));
+                return base.SendAsync(msg, cancel);
+            }
+        }
+
+        private IKubernetes kubernetes;
+        private string group;
+        private string version;
+        private string plural;
+
+
+        public GenericClient(KubernetesClientConfiguration config, string group, string version, string plural)
+        {
+            this.group = group;
+            this.version = version;
+            this.plural = plural;
+
+            if (group.Length == 0)
+            {
+                this.kubernetes = new Kubernetes(config, new DelegatingHandler[] { new TweakApiHandler() });
+            }
+            else
+            {
+                this.kubernetes = new Kubernetes(config);
+            }
+        }
+
+        public async Task<string> ListAsync()
+        {
+            var resp = await this.kubernetes.ListClusterCustomObjectWithHttpMessagesAsync(this.group, this.version, this.plural).ConfigureAwait(false);
+            return resp.Body.ToString();
+        }
+
+        public async Task<string> ListNamespacedAsync(string ns)
+        {
+            var resp = await this.kubernetes.ListNamespacedCustomObjectWithHttpMessagesAsync(this.group, this.version, ns, this.plural).ConfigureAwait(false);
+            return resp.Body.ToString();
+        }
+
+        public async Task<string> ReadNamespacedAsync(string ns, string name)
```

Comment:
`GetXX` would be better I think
align with generated apis

Reply:
I can switch to that, but most of our methods (e.g. `ReadNamespacedPod(...)`) use `Read...`